clcache.py

```diff
@@ -1293,6 +1293,12 @@ def analyze(cmdline):
 
 def invokeRealCompiler(compilerBinary, cmdLine, captureOutput=False, outputAsString=True, environment=None):
     realCmdline = [compilerBinary] + cmdLine
+
+    sourceFile = realCmdline[-1]
+    if '/Tc' + sourceFile in realCmdline or '/Tp' + sourceFile in realCmdline:
+        printTraceStatement("Removing last argument becuase of /Tc (Issue #289)")
+        realCmdline = realCmdline[:-1]
```

Comment:
Is it really safe to assume that the last argument is always a source file argument? I wonder whether maybe this fix should be done on the caller side - but it's not clear to me on first glance where clcache would implicitly add a source file argument to the command line.

Reply:
> Is it really safe to assume that the last argument is always a source file argument?

No. This is an example that fixes the problem but I had hoped that you would have a suggestion about a better fix.

> I wonder whether maybe this fix should be done on the caller side

Yes. Do you have any pointers for where to look?